packages/react/src/IDE/IDE.tsx

```diff
@@ -61,25 +105,77 @@ export type IDEProps = {
+    const [chatAnimationIsDone, setChatAnimationIsDone] = useState(false)
+    const [editorAnimationIsDone, setEditorAnimationIsDone] = useState(false)
+    const chatRef = useRef<IDEChatHandle>(null)
+    const editorRef = useRef<IDEEditorHandle>(null)
+
+    useEffect(() => {
+      if (chatAnimationIsDone && editorAnimationIsDone) {
+        setAnimationIsPaused(false)
+      }
+    }, [chatAnimationIsDone, editorAnimationIsDone])
+
     const childrenArray = useMemo(() => Children.toArray(children), [children])
 
     const ChatChild = childrenArray.find(child => isValidElement(child) && child.type === IDE.Chat)
+    const hasChatChild = !!ChatChild
+    const ClonedChatChild = hasChatChild
+      ? React.cloneElement(ChatChild as React.ReactElement, {
+          ref: chatRef,
+          'data-testid': testId || testIds.chat,
+          animationIsPaused,
+          setAnimationIsDone: setChatAnimationIsDone,
+        })
+      : null
 
     const EditorChild = childrenArray.find(child => isValidElement(child) && child.type === IDE.Editor)
+    const hasEditorChild = !!EditorChild
+    const ClonedEditorChild = hasEditorChild
+      ? React.cloneElement(EditorChild as React.ReactElement, {
+          ref: editorRef,
+          'data-testid': testId || testIds.editor,
+          animationIsPaused,
+          setAnimationIsDone: setEditorAnimationIsDone,
+        })
+      : null
+
+    // If a child exists, we need to account for its animation state
+    const allAnimationsAreDone = (!hasChatChild || chatAnimationIsDone) && (!hasEditorChild || editorAnimationIsDone)
+
     return (
       <section
         data-testid={testId || testIds.root}
-        className={clsx(styles[`IDE--${variant}`], ChatChild && EditorChild && styles['IDE--full-exp'], className)}
+        className={clsx(
+          styles[`IDE--${variant}`],
+          hasChatChild && hasEditorChild && styles['IDE--full-exp'],
+          hasChatChild && !hasEditorChild && styles['IDE--chat-only'],
+          className,
+        )}
       >
         <div {...rest}>
           <div
             className={styles['IDE__inner']}
             style={{['--brand-IDE-height' as string]: height ? `${height}px` : undefined}}
           >
             <div className={styles.IDE__main}>
-              {ChatChild && <>{ChatChild}</>}
-              {EditorChild && <>{EditorChild}</>}
+              {ClonedChatChild}
+              {ClonedEditorChild}
             </div>
+            <PlayPauseButton
+              isPlaying={!animationIsPaused && !allAnimationsAreDone}
+              onPlayPause={isPlaying => {
+                if (allAnimationsAreDone) {
+                  setChatAnimationIsDone(false)
+                  setEditorAnimationIsDone(false)
+                  chatRef.current?.resetAnimation()
+                  editorRef.current?.resetAnimation()
+                } else {
+                  setAnimationIsPaused(!isPlaying)
+                }
+              }}
```

Comment:
could you extract this into a separate handler function please, to avoid recreating it as an inline function. E.g. handleOnPlayPause.